src/main/scala/zio/prelude/NonEmptyList.scala

```diff
@@ -57,11 +57,10 @@ sealed trait NonEmptyList[+A] { self =>
   /**
    * Removes duplicate elements from this `NonEmptyList`.
    */
-  final def distinct(implicit A: Hash[A]): NonEmptyList[A] =
-    reduceMapLeft(a => (single(a), Set(a.hash))) {
+  final def distinct: NonEmptyList[A] =
+    reduceMapLeft(a => (single(a), Set(a))) {
```

Comment:
@adamgfraser look here, there's the crucial bit. The old implementation stores only hashes in the Set, so it can't deal with hash collisions.